src/components/CreateAccountPage.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Button, FormControl, FormGroup } from 'react-bootstrap'
+import { Link } from 'react-router-dom'
+
+/* global fetch */
+
+export default class LogInPage extends Component {
+  constructor (props) {
+    super(props)
+
+    this.getValidationState = this.getValidationState.bind(this)
+    this.checkValidationState = this.checkValidationState.bind(this)
+    this.handleLogInClick = this.handleLogInClick.bind(this)
+    this.createAccount = this.createAccount.bind(this)
+  }
+
+  getValidationState () {
+    let checkInfo = this.props.accountName
+
+    let found = Object.keys(this.props.dbAccounts).findIndex((name) => {
+      return name === checkInfo
+    })
+
+    if (found !== -1 || checkInfo.length < 3) {
+      return 'warning'
+    } else if (found === -1) {
+      return 'success'
+    }
+  }
+
+  checkValidationState () {
+    if (this.getValidationState() === 'warning') {
+      return true
+    } else {
+      return false
+    }
+  }
+
+  createAccount () {
+    let params = {
+      method: 'PUT',
+      body: JSON.stringify({createdAt: new Date(), characters: ['no characters']})
+    }
+
+    fetch(`https://csm-5e.firebaseio.com/users/${this.props.accountName}.json`, params)
+      .then((response) => {
+        return response.json()
+      })
+      .catch((error) => {
+        console.log(error)
+      })
+  }
+
+  handleLogInClick () {
+    this.createAccount()
+    this.props.updateActiveAccount(this.props.accountName)
+  }
+
+  componentWillMount () {
+    fetch('https://csm-5e.firebaseio.com/users.json')
```

Comment:
seems to me that `App.js` should be doing this call and passing the list of users down to these components. Remember the pattern of "Smart and Dumb components" or as Dan Abramov puts it, "Presentational and Container Components". This article might be helpful: https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0